dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/HibernateTrackedEntityStore.java

```diff
@@ -843,145 +795,82 @@ private String getWhereClauseFromFilterConditions(
-            .append(EV_STATUS)
-            .append(EQUALS)
-            .append(SINGLE_QUOTE)
-            .append(EventStatus.SKIPPED.name())
-            .append(SINGLE_QUOTE)
-            .append(SPACE);
-      }
+      sql.append(whereHlp.whereAnd());
+      addEventDateRange(sql, sqlParameters, params);
+      sql.append(whereHlp.whereAnd());
+      addEventStatus(sql, sqlParameters, params);
     }
 
     if (params.hasProgramStage()) {
-      events
-          .append(whereHlp.whereAnd())
-          .append("EV.programstageid = ")
-          .append(params.getProgramStage().getId())
-          .append(SPACE);
+      sql.append(whereHlp.whereAnd()).append("ev.programstageid = :programStageId ");
+      sqlParameters.addValue("programStageId", params.getProgramStage().getId());
     }
 
     if (AssignedUserSelectionMode.NONE == params.getAssignedUserQueryParam().getMode()) {
-      events.append(whereHlp.whereAnd()).append("EV.assigneduserid IS NULL ");
+      sql.append(whereHlp.whereAnd()).append("ev.assigneduserid is null ");
     }
 
     if (AssignedUserSelectionMode.ANY == params.getAssignedUserQueryParam().getMode()) {
-      events.append(whereHlp.whereAnd()).append("EV.assigneduserid IS NOT NULL ");
+      sql.append(whereHlp.whereAnd()).append("ev.assigneduserid is not null ");
     }
 
     if (!params.isIncludeDeleted()) {
-      events.append(whereHlp.whereAnd()).append("EV.deleted IS FALSE");
+      sql.append(whereHlp.whereAnd()).append("ev.deleted is false");
+    }
+  }
+
+  private void addEventDateRange(
+      StringBuilder sql, MapSqlParameterSource sqlParameters, TrackedEntityQueryParams params) {
+    sql.append(
+        switch (params.getEventStatus()) {
+          case COMPLETED, VISITED, ACTIVE ->
+              "ev.occurreddate >= :eventStartDate and ev.occurreddate <= :eventEndDate";
+          case SCHEDULE, OVERDUE, SKIPPED ->
+              "ev.scheduleddate >= :eventStartDate and ev.scheduleddate <= :eventEndDate";
+        });
+    sqlParameters.addValue("eventStartDate", timestampParameter(params.getEventStartDate()));
+    sqlParameters.addValue("eventEndDate", timestampParameter(params.getEventEndDate()));
+  }
+
+  private void addEventStatus(
+      StringBuilder sql, MapSqlParameterSource sqlParameters, TrackedEntityQueryParams params) {
+    if (params.isEventStatus(EventStatus.COMPLETED)) {
+      sql.append("ev.status = :eventStatus");
+      sqlParameters.addValue("eventStatus", EventStatus.COMPLETED.name());
+    } else if (params.isEventStatus(EventStatus.VISITED)
+        || params.isEventStatus(EventStatus.ACTIVE)) {
+      sql.append("ev.status = :eventStatus");
+      sqlParameters.addValue("eventStatus", EventStatus.ACTIVE.name());
+    } else if (params.isEventStatus(EventStatus.SKIPPED)) {
+      sql.append("ev.status = :eventStatus");
+      sqlParameters.addValue("eventStatus", EventStatus.SKIPPED.name());
+    } else if (params.isEventStatus(EventStatus.SCHEDULE)
```

Comment:
I don't think we can merge `EventStatus.SCHEDULE `and `EventStatus.OVERDUE`.
Although I agree it looks same "logically". 😄 

SCHEDULE events are events that has it's scheduledDate in the future.. ( now() <= scheduledDate)
OVERDUE events are events that has it's scheduledDate in the past..  ( now() > scheduledDate)

Did that make sense or am I missing something obvious?